src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -1036,6 +1036,94 @@ public virtual Task<bool> TryUpdateModelAsync<TModel>([NotNull] TModel model)
         }
 
         /// <summary>
+        /// Updates the specified <paramref name="model"/> instance using values from the controller's current
+        /// <see cref="IValueProvider"/> and a <paramref name="prefix"/>.
+        /// </summary>
+        /// <param name="model">The model instance to update.</param>
+        /// <param name="modelType">The type of model instance to update.</param>
+        /// <param name="prefix">The prefix to use when looking up values in the current <see cref="IValueProvider"/>
+        /// </param>
+        /// <returns>A <see cref="Task"/> that on completion returns <c>true</c> if the update is successful</returns>
+        [NonAction]
+        public virtual async Task<bool> TryUpdateModelAsync([NotNull] object model,
+                                                            [NotNull] Type modelType,
+                                                            [NotNull] string prefix)
+        {
+            if (!modelType.IsAssignableFrom(model.GetType()))
+            {
+                var message = Resources.FormatModelType_WrongType(
+                    model.GetType().FullName,
+                    modelType.FullName);
+                throw new ArgumentException(message, nameof(modelType));
+            }
+
+            if (BindingContext == null)
+            {
+                var message = Resources.FormatPropertyOfTypeCannotBeNull(
+                    nameof(BindingContext),
+                    typeof(Controller).FullName);
+                throw new InvalidOperationException(message);
+            }
+
+            return await ModelBindingHelper.TryUpdateModelAsync(
+                model,
+                modelType,
+                prefix,
+                ActionContext.HttpContext,
+                ModelState,
+                MetadataProvider,
+                BindingContext.ModelBinder,
+                BindingContext.ValueProvider,
+                BindingContext.ValidatorProvider);
+        }
+
+        /// <summary>
+        /// Updates the specified <paramref name="model"/> instance using values from the controller's current
+        /// <see cref="IValueProvider"/> and a <paramref name="prefix"/>.
+        /// </summary>
+        /// <param name="model">The model instance to update.</param>
+        /// <param name="modelType">The type of model instance to update.</param>
+        /// <param name="prefix">The prefix to use when looking up values in the current <see cref="IValueProvider"/>.
+        /// </param>
+        /// <param name="predicate">A predicate which can be used to filter properties at runtime.</param>
+        /// <returns>A <see cref="Task"/> that on completion returns <c>true</c> if the update is successful</returns>
+        [NonAction]
+        public async Task<bool> TryUpdateModelAsync(
```

Comment:
I think there should be an overload which can take in a a value provider. 
I would start looking at this from a perspective of, lets provide an overload which lets you provide everything.  And then see what other overloads make sense. ( I am not in favor of adding all of them, but one which has everything should be there).